src/core/model/XojPage.cpp

```diff
@@ -36,80 +36,67 @@ void XojPage::addLayer(Layer* layer) {
     this->currentLayer = npos;
 }
 
-void XojPage::insertLayer(Layer* layer, int index) {
-    if (index >= static_cast<int>(this->layer.size())) {
+void XojPage::insertLayer(Layer* layer, size_t index) {
+    if (index >= this->layer.size()) {
         addLayer(layer);
         return;
     }
 
-    this->layer.insert(this->layer.begin() + index, layer);
+    this->layer.insert(std::next(this->layer.begin(), static_cast<ptrdiff_t>(index)), layer);
     this->currentLayer = index + 1;
 }
 
-void XojPage::removeLayer(Layer* layer) {
-    for (unsigned int i = 0; i < this->layer.size(); i++) {
-        if (layer == this->layer[i]) {
-            this->layer.erase(this->layer.begin() + i);
-            break;
-        }
+void XojPage::removeLayer(Layer* l) {
+    if (auto it = std::find(layer.begin(), layer.end(), l); it != layer.end()) {
+        this->layer.erase(it);
```

Comment:
That's dangerous. Both layer.end() and it are invalidated after erase. Therefore, you must use the iterator returning from erase. But there is an even better solution:

cont.erase + std::remove(if):

```suggestion
    auto const& endi = end(layer);
    this->layer.erase(std::remove(begin(layer),endi, l),endi);
```